src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+        // the ground. The basic idea is that, given two transceivers A, B and a line g representing
+        // the ground. We now would like to determine the length of the shortest path from A to B
+        // via a point on the line. To do this, you can can reflect one of the transceivers B
+        // along the ground g to construct B'. The intersection of the line A-B' with g will result
+        // in the reflection point C in g, and this line A-C-B' now is equal to the shortest path
+        // A-C-B. By then calculating some right angles between A, B', and g, we eventually have a
+        // simple pythagorean triangle where the hypotenuse corresponds to A-C-B', which we can
+        // calculate trivially.
+        let transceiver_along_ground_distance = self.antenna_along_ground_distance();
+        let path_in_air: Length = ((a + b) * (a + b)
+            + transceiver_along_ground_distance * transceiver_along_ground_distance)
+            .sqrt();
+        let shortest_path_length: Length =
+            path_in_air + self.transmitter.electric_length + self.receiver.electric_length;
+
+        // At this point we've determined the length of the shortest path between the two
+        // transceivers. We now convert it into a travel time.
+
+        // It would be nice if this could be a const, but unfortunately the Velocity constructor is
+        // too complicated for what Rust deems acceptable for a const.
+        let speed_of_light: Velocity = Velocity::new::<meter_per_second>(299_792_458.);
+
+        // While radio waves propagate at the speed of light in vacuum, they travel slower in other
+        // mediums. This correction is called refractive index.
+        // If you want, you can try calculating yourself using
+        // https://www.fig.net/resources/proceedings/fig_proceedings/fig_2002/Js28/JS28_rueger.pdf
+        // However as the index varies by only around 0.01%, the effects on the final radio height
+        // as measured by civil aircraft is negligible.
+        let refractive_radio_index = Ratio::new::<ratio>(1.000293);
+        let speed_of_radio_waves = speed_of_light / refractive_radio_index;
+
+        let travel_time = shortest_path_length / speed_of_radio_waves;
+
+        if shortest_path_length < Length::new::<foot>(10200.) {
+            Some(TransceiverPairResponse::new(travel_time))
+        } else {
+            None
+        }
+    }
+}
+
+impl SimulationElement for Ala52BTransceiverPair {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.alt_above_ground = Length::new::<foot>(reader.read(&self.alt_above_ground_id));
+        self.pitch = Angle::new::<degree>(reader.read(&self.pitch_id));
+        self.bank = Angle::new::<degree>(reader.read(&self.bank_id));
+    }
+}
+
+pub trait RadioAltimeter {
+    fn radio_altitude(&self) -> Arinc429Word<Length>;
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Ala52BAircraftInstallationDelay {
```

Comment:
**thought**: Adding a comment as to what this represents might be helpful. I'm not quite able to determine it based on the usage. (Maybe `Delay` feels like something of `Time`, but it's `Length` here, which confuses me a bit 😀.

Reply:
Yeah, it's a bit confusing, but I'm just following the industry here. Apparently Aircraft Installation Delay / AID is standard terminology (presumably defined in ARINC 707), and the pins are also defined in terms of Lengths.

See for example [this manual](https://www.viavisolutions.com/ja-jp/literature/alt-8000-operations-manual-manuals-user-guides-en.pdf) (search for "AID") and [this book](https://soaneemrana.org/onewebmedia/AIRCRAFT%20RADIO%20SYSTEM%20BY%20J%20POWEL.pdf):

> Part of the two-way travel time is accounted for by the aircraft installation delay (AID). Since, ideally,
the radio altimeter should read zero feet on touchdown, the residual altitude which accounts for the AID is made up of cable length multiplied by a factor (typically 1.5) to allow for propagation speed, and the sum of the antenna heights. 

Tracer already raised the cable propagation speed further up which I didn't want to model, but I don't think that changes anything about the interface.